src/lib/net.js

```diff
@@ -681,3 +681,65 @@ foam.CLASS({
     }
   ]
 });
+
+
+foam.ENUM({
+  package: 'foam.net',
+  name: 'Environment',
+
+  values: [
+    { name: 'BROWSER', label: 'Browser' },
+    { name: 'NODE',    label: 'Node'    }
+  ]
+});
+
+
+foam.CLASS({
+  package: 'foam.net',
+  name: 'Universal',
+
+  axioms: [ foam.pattern.Singleton.create() ],
+  requires: [ 'foam.net.Environment' ],
+
+  properties: [
+    {
+      class: 'Enum',
+      of: 'foam.net.Environment',
+      name: 'environment',
+      factory: function() {
+        return typeof process === 'undefined' ?
+            this.Environment.BROWSER :
+            this.Environment.NODE;
+      }
+    }
+  ],
+
+  methods: [
+    function getRuntimeClass(cls) {
+        return foam.lookup((this.environment === this.Environment.NODE ?
+            'foam.net.node' : 'foam.net') + '.' + cls.name);
+    },
+    function installInClass(cls) {
+      var runtimeCls = this.getRuntimeClass(cls);
+      cls.create = function(args, opt_parent) {
+        return runtimeCls.create(args, opt_parent);
+      };
+    }
+  ]
+});
+
+
+foam.CLASS({
```

Comment:
TODO: Just register the appropriate model(s) under `universal` package name, statically. (using `foam.register()`).